hal/imx_rt.ld

```diff
@@ -1,7 +1,7 @@
 /* Specify the memory areas */
 MEMORY
 {
-  FLASH(rx)  : ORIGIN = 0x60000000, LENGTH = @BOOTLOADER_PARTITION_SIZE@
+  FLASH(rx)  : ORIGIN = @ARCH_FLASH_OFFSET@, LENGTH = @BOOTLOADER_PARTITION_SIZE@
```

Comment:
@dgarske I referenced this in the description, but let me know if this (and the corresponding `sed` command in the top-level Makefile) isn't okay for some reason, or breaks anything. The issue is that the build infra assumes one linker script per target, but the RT1064 has a slightly different memory map than the RT1050 and RT1060. Specifically, the RT1064 uses `FlexSPI2` to access the in-package QSPI flash (which it boots from, and is mapped to `0x7000_0000`), whereas the 1050 and 1060 use FlexSpi1 (mapped to `0x6000_0000`). Let me know if this is too cursed and I can find a better way.

Reply:
This is the cleanest way to do this